ic-stable-storage/src/log.rs

```diff
@@ -0,0 +1,227 @@
+use std::marker::PhantomData;
+use std::mem::size_of;
+use std::rc::Rc;
+
+use candid::{CandidType, Deserialize};
+
+use super::error::Result;
+use super::{
+    from_bytes, to_byte_vec, Memory, RestrictedMemory, StableBTreeMap, StableMemory, VirtualMemory, PADDING
+};
+
+type Mem<const INDEX: u8> = VirtualMemory<Rc<RestrictedMemory<StableMemory>>, INDEX>;
```

Comment:
Do we really need to restrict stable log and map to use only `VirtualMemory`. I don't see any reason, why they cannot use any kind of memory provided for them. If it's only for convinient initialization, it can be done through a different `new` method. One would take memory as an argument, and the other would initialize `VirtualMemory` on its own.

Reply:
These types should only operate on `VirtualMemory`.
If we mixed this with `StableMemory` it would probably corrupt the storage.

It would also make the API more verbose for no real gain.

If there is a use case where we should use any of these types without the virtual memory we should discuss it first as it could have quite the impact on what's already used.